src/services/roomservice/index.ts

```diff
@@ -37,7 +44,9 @@ export const routes = (app: Application) => {
     '/rooms',
     authMiddleware,
     asyncHandler(async (req: Request, res: Response) =>
-      res.json(await getRooms(req.query.rentalId as string))
+      res.json(
+        await getRooms(req.query.rentalId as string, (req.query.isShared as unknown) as boolean)
```

Comment:
`req.query.isShared` is not a boolean but a string (and it could be any string.) 

Instead it would be best to check if isShared is one of `["true", "false", undefined]` and return a 400 Bad Request if not. Then pass it as a string to getRooms (or convert to a boolean.)